File: 1328.破坏回文串.go
/*
 * @lc app=leetcode.cn id=1328 lang=golang
 *
 * [1328] 破坏回文串
 */
package algorithmcamp

// @lc code=start
func breakPalindrome(palindrome string) string {
	n := len(palindrome)
	if n <= 1 {
		return ""
	}
	for i := 0; i+i < n-1; i++ {
		if palindrome[i] != 'a' {
			return palindrome[:i] + "a" + palindrome[i+1:]
		}
	}
	return palindrome[:n-1] + "b"
}

// @lc code=end

File: 1328.破坏回文串.rs
/*
 * @lc app=leetcode.cn id=1328 lang=rust
 *
 * [1328] 破坏回文串
 */

// @lc code=start
impl Solution {
    pub fn break_palindrome(palindrome: String) -> String {
        if palindrome.len() < 2 {
            return String::new();
        }
        let mut s: Vec<char> = palindrome.chars().collect();
        let n = palindrome.len();
        for i in 0..n / 2 {
            if s[i] != 'a' {
                s[i] = 'a';
                return s.into_iter().collect();
            }
        }
        s[n - 1] = 'b';
        s.into_iter().collect()
    }
}
// @lc code=end
